app/models/user.rb

```diff
@@ -1,7 +1,16 @@
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable and :omniauthable
+
+  enum role: [:standard, :premium, :admin]
+  after_initialize :set_default_role, :if => :new_record?
+
+  def set_default_role
+    self.role ||= :standard
+  end
+
+  has_many :wikis #, dependent: :destroy
```

Comment:
Curious, why you commented the dependent: destroy out?

Reply:
I was testing a bunch of different configurations in order to see the results and thus better understand the subject matter, and simply didnt get back to uncomment this code.